.github/workflows/dispatch-update-to-build-ci.yml

```
@@ -0,0 +1,48 @@
+name: "Dispatch Event to Build CI"
+on:
+  push:
+    branches:
+      - main
+    tags: 
+      - '*.*.*'
+    paths:
+      - '**.py'
+
+jobs:
+  send-tag-update-to-build-ci:
+    name: Send Tag Update to Build CI repo
+    runs-on: ubuntu-latest
+    if: github.ref_type == 'tag'
+    steps:
+      - uses: actions/checkout@v3
+      
+      - name: Get names of all package directories
+        id: directories
+        run: |
+          # list all the packages, turn it into a space-separated list!
+          echo "modified=$(ls packages | tr '\n' ' ')" >> $GITHUB_OUTPUT
+
+      - name: Send event and tag to build-ci repo
+        run: |
+          curl -XPOST -u "${{ secrets.RELEASE_BUILD_CI_PAT_USERNAME }}:${{ secrets.RELEASE_BUILD_CI_PAT }}" \
+            -H "Accept: application/vnd.github.everest-preview+json" \
+            -H "Content-Type: application/json" https://api.github.com/repos/ACCESS-NRI/build-ci/actions/workflows/notified-of-spack-packages-update.yml/dispatches \
+            --data '{"ref": "main", "inputs":{"spack-packages-version": "${{ github.ref_name }}", "packages": "${{ steps.directories.outputs.modified }}"}}'
+
+  send-branch-update-to-build-ci:
+    name: Send Branch Update to Build CI repo
+    runs-on: ubuntu-latest
+    if: github.ref_type == 'branch'
+    steps:
+      - name: Get names of directories modified in push/tag
+        id: directories
+        run: |
+          # get all the changed files in the push, only keep the parent directory (corresponds to the package name), and turn it into a space-separated list!
+          echo "modified=$(git diff --name-only ${{ github.event.after }} ${{ github.event.before }} | cut -d '/' -f 2 | tr '\n' ' ')" >> $GITHUB_OUTPUT
```

Should this snippet be a stand-alone shell script that we could test? Possible a separate shelll script within the repository that could be changed depending on circumstances.

Better to have a test for the script itself than diagnosing down the track?

---

Good point. Although, are Github contexts (`${{ github.whatever }}`) available from within the script?
From a cursory glance around, it's not accessible, so people pass in the context as env vars instead. But, we have secrets in that curl request (`${{ secrets.RELEASE_BUILD_CI_PAT(_USERNAME) }}`) which could get exposed through passing them as env vars. I'm leaning against it if this is the case.

---

True. Maybe for further dicussion down the track.

---

I think the stand-alone snippet suggestion just applied to the shell magic to extract the modified directories. This might involve making a script like this (named `./util/find_modified_packages.sh`):
```
#!/bin/bash
git diff --name-only $1 $2 | cut -d '/' -f 2 | tr '\n' ' '
```
and changing the CI command to
```suggestion
          echo "modified=$(./util/find_modified_packages.sh ${{ github.event.after }} ${{ github.event.before }} )" >> $GITHUB_OUTPUT
```
and add CI to test `./util/find_modified_packages.sh`